src/concrete/ml/sklearn/linear_model.py

```diff
@@ -687,13 +680,7 @@ def fit(  # type: ignore[override]
 
         # If the model should be trained using FHE training
         if self.fit_encrypted:
-            if fhe is None:
-                fhe = "disable"
-                warnings.warn(
```

Comment:
no this warning was made on purpose, I think we should keep it no ? the idea if that users should set `fhe` to something if they activate `fit_encrypted` to be sure they know what they are doing

the reason why `fhe` is not defaulted to `disable` is to avoid the ambiguous situation of having `fit_encrypted` to false and `fhe` to `disable`, which wouldn't make much sense since training would be done in floating points with sklearn

Reply:
Maybe I miss something but this didn't really make sense to me. If we have a warning here then why not for all our models in the predict?

> avoid the ambiguous situation of having fit_encrypted to false and fhe to disable, which wouldn't make much sense since training would be done in floating points with sklearn

If fit_encrypted = False then [we have this](https://github.com/zama-ai/concrete-ml/blob/4f759790ab77d25f296bfa41626b3fa0a80021bc/src/concrete/ml/sklearn/linear_model.py#L705-L710)

Or what do I miss?

Reply:
ah yes you right about the second case, good thing then !

but the reason we don't have this in the `predict` methods is simply because there is not ambiguity there, we only have one parameter related to fhe execution : `fhe`. Whereas with encrypted training, we have this additional `fit_encrypted` in the init.

so more or less I believe the idea here was to better determine the `fhe` parameter's role, by having something like : 
- `None` : float clear
- `disable | simulate | fhe ` : the usual

so when a user sets `fit_encrypted`, it's better to make sure he's aware of what mode he's using

Reply:
There is no notion of fhe= if it's a training in the clear and we have a assert on this. To me, fhe="disable" should be the default if users use fit_encrypted = True.

I don't think that the warning ` "Parameter 'fhe' isn't set while FHE training is enabled.\n Defaulting to '{fhe=}'"` is of any help to the user.

If the API is really confusing, having warning to try to explain isn't the right direction. Instead we should rethink the API.

Reply:
well iirc there was some discussion on this in particular when the API was made and this is how we decided to go for, maybe @fd0r can tell us more about it !

Reply:
Yes @RomanBredehoft got it right.

We don't want `disable` by default since it wouldn't make sense if `fit_encrypted=False`.
But not providing a value for `fhe` if `fit_encrypted=True` is also "dangerous". The idea here was to make sure that the user  has to provide some parameter for `fhe` in this case.

Now that I'm re-reading that I would have put an Exception instead of a warning here.

I don't think we should remove the warning.

Reply:
> We don't want disable by default since it wouldn't make sense if fit_encrypted=False.

But if `fit_encrypted=False.` and fhe != None, we raise an error so this is not possible.


> But not providing a value for fhe if fit_encrypted=True is also "dangerous"

I really don't understand why this is "dangerous". For the inference, disable is the default. Why would it be dangerous to do it for training?


Also, what should the user learn from  `"Parameter 'fhe' isn't set while FHE training is enabled.\n Defaulting to '{fhe=}'"`?

Reply:
> I really don't understand why this is "dangerous". For the inference, disable is the default. Why would it be dangerous to do it for training?

Because we would be changing the default from the signature mostly

Reply:
> Also, what should the user learn from "Parameter 'fhe' isn't set while FHE training is enabled.\n Defaulting to '{fhe=}'"?

Mainly that the default value is changing is shouldn't really  happen. And that the prefered way is to specify this argument.

Reply:
Alright 🏳️ well I will add back the warnings and tests for it.